tensorboard/plugins/debugger/debugger_server_lib.py

```diff
@@ -0,0 +1,321 @@
+               receive_port,
+               logdir,
+               always_flush=False):
+    """Receives health pills from a debugger and writes them to disk.
+
+    Args:
+      receive_port: The port at which to receive health pills from the
+        TensorFlow debugger.
+      logdir: The directory in which to write events files that TensorBoard will
+        read. The ~ in this string might not be expanded yet.
+      always_flush: A boolean indicating whether the EventsWriter will be
+        flushed after every write. Can be used for testing.
+    """
+    # We create a special directory within logdir to store debugger-related
+    # events (if that directory does not already exist). This is necessary
+    # because for each directory within logdir, TensorBoard only reads through
+    # each events file once. There may be other non-debugger events files being
+    # written to at the same time. Without this special directory, TensorBoard
+    # may stop surfacing health pills after some arbitrary step value.
+    debugger_directory = os.path.join(
+        os.path.expanduser(logdir), constants.DEBUGGER_DATA_DIRECTORY_NAME)
+
+    if not tf.gfile.Exists(debugger_directory):
+      try:
+        tf.gfile.MakeDirs(debugger_directory)
+        tf.logging.info("Created directory for debugger data: %s",
+                        debugger_directory)
+      except tf.OpError as e:
+        tf.logging.error(
+            "Could not make directory for debugger data: %s. Error: %s",
+            debugger_directory, e)
+
+    self._events_writer_manager = events_writer_manager_lib.EventsWriterManager(
+        events_directory=debugger_directory,
+        always_flush=always_flush)
+
+    # Write an event with a file version as the first event within the events
+    # file. If the event version is 2, TensorBoard uses a path for purging
+    # events that does not depend on step. This is important because debugger
+    # events use a notion of step that differs from that of the rest of
+    # TensorBoard.
+    try:
+      self._events_writer_manager.write_event(
+          tf.Event(
+              wall_time=0, step=0, file_version=constants.EVENTS_VERSION))
+    except IOError as e:
+      tf.logging.error(
+          "Writing to %s failed: %s",
+          self._events_writer_manager.get_current_file_name(), e)
+
+    # See if a backup file exists. If so, use it to initialize the registry.
+    self._registry_backup_file_path = os.path.join(
+        debugger_directory, constants.ALERT_REGISTRY_BACKUP_FILE_NAME)
+    initial_data = None
+
+    if tf.gfile.Exists(self._registry_backup_file_path):
+      # A backup file exists. Read its contents to use for initialization.
+      with tf.gfile.Open(self._registry_backup_file_path, "r") as backup_file:
+        backup_data_json_string = backup_file.read().strip()
+
+        try:
+          # Use the data to initialize the registry.
+          initial_data = json.loads(backup_data_json_string)
```

Comment:
You should be able to replace this and the definition above with `initial_data = json.load(backup_file)`.